CHANGELOG.md

```diff
@@ -116,6 +116,7 @@
 - [8945](https://github.com/vegaprotocol/vega/issues/8945) - Expose missing order from the stop order `GraphQL` endpoint.
 - [9034](https://github.com/vegaprotocol/vega/issues/9034) - Add missing transfer types to `GraphQL` schema definition.
 - [9075](https://github.com/vegaprotocol/vega/issues/9075) - `Oracle Specs API` can fail when oracle data is null.
+- [8944](https://github.com/vegaprotocol/vega/issues/8944) - Fix ignoring of asset `ID` in ledger export, and make it optional.
```

Comment:
I think this is also a breaking change right? because the type in the gRPC was a `string` but nows its a `*string`. So can we make sure we include it in the right changelog section and add the relevant labels to the PR and ticket?

Reply:
the previous behaviour was that you had to provide an asset ID, but it ignored it and gave you all the assets anyway :)

Reply:
Its breaking in the sense the the gRPC is no longer backwards-compatible, i.e someone wanted to _start_ using `ExportLedgerEntries` but hadn't yet update their client-side-generate-from-the-protos, they'll get an unmarshaling error when their message hits the datanode because the type has changed.

Equally the person who submitted the bug might be like "its fixed now, I can just re-run my script" and they'll get an unmarshalling error too. So from their point-of-view its going from "I supply an assetId and it doesn't filter" to "now it doesn't work at all".

Reply:
ah I see, I initially thought you meant breaking as in breaking usage of the API

You mean breaking in the sense that the generated go bindings for grpc are no longer compatible? because I think they actually would be; it's my understanding protobuf wire format doesn't change by making a field optional; the zero value and a non-present optional are both (as I understand it) encoded as simply that record not being present in the encoded message..

 though I can't find any definitive source to back that up [this post](https://github.com/golang/protobuf/issues/225) touches upon it

Reply:
Ok, well if thats true then its fine as it is!